logux-vue-todomvc/src/App.vue

```diff
@@ -0,0 +1,406 @@
+<template>
+  <section class="todoapp">
+    <!-- header -->
+    <header class="header">
+      <h1>todos</h1>
+      <input class="new-todo"
+        autofocus
+        autocomplete="off"
+        placeholder="What needs to be done?"
+        @keyup.enter="addTodo">
+    </header>
+    <!-- main section -->
+    <section class="main" v-show="todos.length">
+      <input class="toggle-all" id="toggle-all"
+        type="checkbox"
+        :checked="allChecked"
+        @change="toggleAll(!allChecked)">
+      <label for="toggle-all"></label>
+      <ul class="todo-list">
+        <TodoItem
+          v-for="(todo, index) in todos"
+          :key="index"
+          :todo="todo"
+        />
+      </ul>
+    </section>
+    <!-- footer -->
+    <footer class="footer" v-show="todos.length">
+      <span class="todo-count">
+        <strong>{{ remaining }}</strong>
+        {{ remaining | pluralize('item') }} left
+      </span>
+    </footer>
+  </section>
+</template>
+
+<script>
+import TodoItem from './components/TodoItem.vue'
+
+export default {
+  components: { TodoItem },
+  data () {
+    return {}
+  },
+  computed: {
+    todos () {
+      return this.$store.state.todos
+    },
+    allChecked () {
+      return this.todos.every(todo => todo.done)
+    },
+    remaining () {
+      return this.todos.filter(todo => !todo.done).length
+    }
+  },
+  methods: {
+	toggleAll () {
+		this.todos
+			.forEach(todo => {
+				this.$store.commit.sync({ type: 'toggle', id: todo.id, done: !todo.done })
+			})
+	},
+    addTodo (e) {
+	  const text = e.target.value
+      if (text.trim()) {
+        this.$store.commit.sync({ type: 'newTodo', text, done: false})
+      }
+      e.target.value = ''
+    }
+  },
+  filters: {
+    pluralize: (n, w) => n === 1 ? w : (w + 's'),
+    capitalize: s => s.charAt(0).toUpperCase() + s.slice(1)
+  }
+}
+</script>
+<style>
+button {
+	margin: 0;
+	padding: 0;
+	border: 0;
+	background: none;
+	font-size: 100%;
+	vertical-align: baseline;
+	font-family: inherit;
+	font-weight: inherit;
+	color: inherit;
+	-webkit-appearance: none;
+	appearance: none;
+	-webkit-font-smoothing: antialiased;
+	-moz-osx-font-smoothing: grayscale;
+}
+body {
+	display: flex;
+	font: 14px 'Helvetica Neue', Helvetica, Arial, sans-serif;
+	line-height: 1.4em;
+	background: #f5f5f5;
+	color: #111111;
+	min-width: 230px;
+	max-width: 550px;
+	margin: 0 auto;
+	-webkit-font-smoothing: antialiased;
+	-moz-osx-font-smoothing: grayscale;
+	font-weight: 300;
+}
+:focus {
+	outline: 0;
+}
+.hidden {
+	display: none;
+}
+.todoapp {
+	width: 720px;
+	flex-direction: column;
+	background: #fff;
+	margin: auto;
+	position: relative;
+	box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2),
+	            0 25px 50px 0 rgba(0, 0, 0, 0.1);
+}
+.todoapp input::-webkit-input-placeholder {
+	font-style: italic;
+	font-weight: 300;
+	color: rgba(0, 0, 0, 0.4);
+}
+.todoapp input::-moz-placeholder {
```

Comment:
You send vendor prefixes to the creator of Autoprefixer? :D

Reply:
actually i just stole css from https://github.com/vuejs/vuex/tree/dev/examples/todomvc

a can polish css but it requires some time

Reply:
Yeap, we should not copy-paste anything without looking inside